packages/nimbus/src/components/dialog/dialog.types.ts

```diff
@@ -0,0 +1,173 @@
+import { type ComponentProps } from "react";
+import { type RecipeVariantProps } from "@chakra-ui/react";
+import { dialogSlotRecipe } from "./dialog.recipe";
+import { type ModalOverlayProps } from "react-aria-components";
+import type { DialogModalOverlaySlotProps } from "./dialog.slots";
+import type { IconButtonProps } from "@/components";
+
+/**
+ * Props for the Dialog.Root component
+ *
+ * The root component that provides context and state management for the dialog.
+ * Uses React Aria's DialogTrigger for accessibility and state management.
+ *
+ * This component handles configuration through recipe variants that are passed
+ * down to child components via context.
+ */
+export interface DialogRootProps
+  extends RecipeVariantProps<typeof dialogSlotRecipe> {
+  /**
+   * The children components (Trigger, Content, etc.)
+   */
+  children: React.ReactNode;
+
+  /**
+   * Whether the dialog is open (controlled mode)
+   */
+  isOpen?: boolean;
+
+  /**
+   * Whether the dialog is open by default (uncontrolled mode)
+   * @default false
+   */
+  defaultOpen?: boolean;
+
+  /**
+   * Whether the dialog can be dismissed by clicking the backdrop or pressing Escape.
+   * If true, clicking outside the dialog or pressing Escape will close it.
+   * @default true
+   */
+  isDismissable?: ModalOverlayProps["isDismissable"];
+
+  /**
+   * Whether keyboard dismissal (Escape key) is disabled.
+   * If true, pressing Escape will NOT close the dialog.
+   * @default false
+   */
+  isKeyboardDismissDisabled?: ModalOverlayProps["isKeyboardDismissDisabled"];
+
+  /**
+   * Function to determine whether the dialog should close when interacting outside.
+   * Receives the event and returns true to allow closing, false to prevent.
+   */
+  shouldCloseOnInteractOutside?: ModalOverlayProps["shouldCloseOnInteractOutside"];
+
+  /**
+   * Callback fired when the dialog open state changes
+   * @param isOpen - Whether the dialog is now open
+   */
+  onOpenChange?: (isOpen: boolean) => void;
+
+  /** A Title for the dialog, optional, as long as there is a Heading
+   * component used inside the Dialog.Content with a `slot`-property set to `title`.
+   */
+  "aria-label"?: string;
+}
+
+/**
+ * Props for the Dialog.Trigger component
+ *
+ * The trigger element that opens the dialog when activated.
+ */
+export interface DialogTriggerProps extends ComponentProps<"button"> {
+  /**
+   * The trigger content
+   */
+  children: React.ReactNode;
+
+  /**
+   * Whether to render as a child element (use children directly as the trigger)
+   * @default false
+   */
+  asChild?: boolean;
```

Comment:
I know we're setting this precedent, but I'm still not sold on this for consumers to understand. We don't have a good internal "detect a RAC button" solution, so it feels like we're pawning this off to the consumer, to understand our abstraction.

It is what it is 🤷‍♂️